### DISCUSSION TITLE
Custom configuration file name doesn't work

### DISCUSSION THREAD

**Discussion by pasqualevitiello** [tailwindlabs/tailwindcss#7158] [Help] [answered]
Trying to use a custom file name for Tailwind config file with PostCSS, like explained in the docs: https://tailwindcss.com/docs/configuration#using-a-different-file-name

For some reason it doesn't work, am I missing anything?

----

**Steps to reproduce the issue:**

- [Install a CRA app](https://tailwindcss.com/docs/guides/create-react-app)

- Rename `tailwind.config.js` to `tailwindcss-config.js`

- Configure template paths:

```
module.exports = {
  content: [
    "./src/**/*.{js,jsx,ts,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

- Edit postcss.config.js:

```
module.exports = {
  plugins: {
    tailwindcss: { config: './tailwindcss-config.js' },
    autoprefixer: {},
  },
}
```

- `npm start`

----

**Result is:**

<img width="1457" alt="Screenshot 2022-01-21 at 16 36 02" src="https://user-images.githubusercontent.com/2683512/150555330-dfc80993-263b-45a4-9838-748d7a9810c8.png">

**Comment by adamwathan** [comment, marked as answer]
This is almost guaranteed to be CRA specific and caused by CRA ignoring most customizations in general. If you want to make customizations I would highly recommend switching to a different tool, CRA will fight you the whole time. Vite would be my recommendation, it does everything CRA does but is easily customized and much faster.
